October news (#405)
ropensci/roweb3: content/blog/2022-10-21-ropensci-news-digest-october-2022/index.Rmd

---
title: rOpenSci News Digest, October 2022
author:
  - The rOpenSci Team
date: '2022-10-21'
slug: ropensci-news-digest-october-2022
categories: []
tags:
  - newsletter

> description: rOpenSci Champions programm call for champions and mentors; R-universe search; coworking

Reviewer: ```suggestion
description: rOpenSci Champions Program call for champions and mentors; R-universe search; coworking
```

params:
  last_newsletter: '2022-09-23'
---

```{r setup, include=FALSE}
library("magrittr")
library("rlang")
last_newsletter <- anytime::anytime(params$last_newsletter)
knitr::opts_chunk$set(echo = FALSE)
url <- sprintf(
    "/blog/%s/%s/%s/%s",
    lubridate::year(rmarkdown::yaml_front_matter(knitr::current_input())$date),
    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### rOpenSci Champions Program: call for Champions and call for Mentors

> [rOpenSci Champions Program](/champions/) is part of a series of activities and projects we are carrying out to ensure our research software serves everyone in our communities. 

Reviewer: ```suggestion
The [rOpenSci Champions Program](/champions/) is part of a series of activities and projects we are carrying out to ensure our research software serves everyone in our communities. 
```

That is why this program focuses on __people who belong to groups that are historically and systematically excluded__ from the open software and research software communities and who are interested in contributing to rOpenSci and the broader open source and research software communities.

#### Call for Champions

If you are selected as a Champion, over the course of a year you will develop a project, get training, and be mentored. You will also receive a small stipend. 

> You can apply until **November 7, 2022** [using a form](/champions/#how-to-apply).  You can read more details in this [blog post](/blog/2022/09/22/launch-champions-program/).

Reviewer: ```suggestion
You can apply until **November 7, 2022** with this [form](/champions/#how-to-apply).  You can read more details in this [blog post](/blog/2022/09/22/launch-champions-program/).
```

#### Call for Mentors

### Better search on R-universe

<!-- to be curated manually -->

### Coworking sessions continue!

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

> * [Tuesday, 01 November 2022 9 AM North American Pacific / 16:00 UTC ](/events/coworking-2022-11/) "Champions Program Application Clinic" *Hosted by rOpenSci community manager [Yanina Bellini Saibene](/author/yanina-bellini-saibene/) and [Steffi LaZerte](/author/steffi-lazerte/)*

Reviewer: ```suggestion
* [Tuesday, November 1st 9 AM North American Pacific / 16:00 UTC ](/events/coworking-2022-11/) "Champions Program Application Clinic" *Hosted by rOpenSci community manager [Yanina Bellini Saibene](/author/yanina-bellini-saibene/) and [Steffi LaZerte](/author/steffi-lazerte/)*
```

>     * Cowork on a project of your choice;
>     * Prepare your application for the Champions Program;
>     * Or talk to [Yanina](/author/yanina-bellini-saibene/) about the application process.

Reviewer: ```suggestion
    * Prepare your application for the Champions Program.
    * Talk to [Yanina](/author/yanina-bellini-saibene/) about the application process.
    * Cowork independently on work related to R. Work on packages that tend to be neglected; What ever you need to get done!
```

    
* [Tuesday, December 6th, 9 AM Australian Western / 1:00 UTC](/events/coworking-2022-12/) "Getting started with targets!" *Hosted by community host [Nick Tierney](/author/nicholas-tierney/) and [Steffi LaZerte](/author/steffi-lazerte/)*

>     - Dive into the world of targets! Do some studying; Start setting it up for some of your projects.
>     - Cowork independently on work related to R. Work on packages that tend to be neglected; What ever you need to get done!

Reviewer: ```suggestion
    - Dive into the world of targets! Do some studying; Start setting it up for some of your projects;
    - Ask [Nick](/author/nicholas-tierney/) for suggestions how how to get started or tips and tricks;
    - Cowork independently on work related to R. Work on packages that tend to be neglected; What ever you need to get done!
```

```{r usecases2, results='asis'}
format_case <- function(usecase) {
  string <- sprintf("* [%s](%s). Reported by %s.", sub("\\.$", "", usecase$title), usecase$url, usecase$reporter)
}
cat(
  paste0(
    purrr::map_chr(usecases, format_case),
    collapse = "\n\n"
  )
)
```

Explore [other use cases](/usecases) and [report your own](https://discuss.ropensci.org/c/usecases/10)!

## Call for maintainers

<!--IF CALL
* [our guidance on _Changing package maintainers_](https://devguide.ropensci.org/changing-maintainers.html)
* [our _Package Curation Policy_](https://devguide.ropensci.org/curationpolicy.html)

IF NO CALL
There are no open calls for new maintainers at this point but you can refer to our [contributing guide](https://contributing.ropensci.org/) for finding ways to get involved!
As the maintainer of an rOpenSci package, feel free to contact us on Slack or email `info@ropensci.org` to get your call for maintainer featured in the next newsletter. -->

## Package development corner

Some useful tips for R package developers. :eyes:

### Are you ready to develop packages?

> If the question applies to _you_ as an R user, please refer to the resources listed in our [dev guide](https://devguide.ropensci.org/building.html#learning-about-package-development).
> Now, in [Susan Johnston’s wise words](https://github.com/susjoh/fibonacci), if you can open R, if you can install a package, if you can write functions or learn how to write functions, you can write an R package!

Reviewer: ```suggestion
If the question applies to _you_ as an R user, know that in [Susan Johnston’s wise words](https://github.com/susjoh/fibonacci), if you can open R, if you can install a package, if you can write functions or learn how to write functions, you can write an R package! If you're still not sure please refer to the resources listed in our [dev guide](https://devguide.ropensci.org/building.html#learning-about-package-development).
```

> If the question applies to _your machine_, you might like:

Reviewer: ```suggestion
On the other hand, if this question applies to _your machine_, you might use the following functions to find out:
```

* [`devtools::dev_sitrep()`](https://devtools.r-lib.org/reference/dev_sitrep.html);

> * [`devtools::has_devel()`](https://rdrr.io/cran/pkgbuild/man/has_compiler.html);

Reviewer: ```suggestion
* [`devtools::has_devel()`](https://r-lib.github.io/pkgbuild/reference/has_compiler.html);
```

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### rOpenSci Champions Program: call for Champions and call for Mentors

The [rOpenSci Champions Program](/champions/) is part of a series of activities and projects we are carrying out to ensure our research software serves everyone in our communities. 
That is why this program focuses on __people who belong to groups that are historically and systematically excluded__ from the open software and research software communities and who are interested in contributing to rOpenSci and the broader open source and research software communities.

#### Call for Champions

If you are selected as a Champion, over the course of a year you will develop a project, get training, and be mentored. You will also receive a small stipend. 

You can apply until **November 7, 2022** with this [form](/champions/#how-to-apply).  You can read more details in this [blog post](/blog/2022/09/22/launch-champions-program/).

#### Call for Mentors

### Introducing R-universe power search

> Earlier this year we announced [r-universe search](https://ropensci.org/blog/2022/03/23/runiverse-search/) which lets you easily browse and discover R packages across the various R ecosystems. We have now upgraded the search system to support advanced search queries, to search for particular properties of an R package.

Reviewer: Just for consistency 😅 
```suggestion
Earlier this year we announced [R-universe search](https://ropensci.org/blog/2022/03/23/runiverse-search/) which lets you easily browse and discover R packages across the various R ecosystems. We have now upgraded the search system to support advanced search queries, to search for particular properties of an R package.
```

{{< tweet user="opencpu" id="1573334441849667584" >}}

Advanced search lets you query packages that specifially match a particular author, maintainer, keyword, or topic. But you can also filter based on metadata derived by the build system, for example packages that contain a particular function or dataset, or packages with a particular contributor or dependency.

To give it a try, simply go to https://r-universe.dev and click the advanced search icon:

[![screenshot of the advanced search box](advanced.png)](https://r-universe.dev/search)

> 

Author: added!

}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### rOpenSci Champions Program: call for Champions and call for Mentors

The [rOpenSci Champions Program](/champions/) is part of a series of activities and projects we are carrying out to ensure our research software serves everyone in our communities. 
That is why this program focuses on __people who belong to groups that are historically and systematically excluded__ from the open software and research software communities and who are interested in contributing to rOpenSci and the broader open source and research software communities.

#### Call for Champions

If you are selected as a Champion, over the course of a year you will develop a project, get training, and be mentored. You will also receive a small stipend. 

You can apply until **November 7, 2022** with this [form](/champions/#how-to-apply).  You can read more details in this [blog post](/blog/2022/09/22/launch-champions-program/).

#### Call for Mentors

> 

Reviewer 2: ```suggestion
We are seeking **mentors** to support our inaugural cohort of rOpenSci Champions.

Mentors will play a vital role in the rOpenSci Champions Program by helping select the first cohort of Champions and advising and inspiring their mentees. Mentors will also __connect__ their mentees to people, programs, and organizations; __recommend__, resources, readings, training, and experiences; and provide __feedback__ on the mentee's project.
 
[Learn the details](/blog/2022/10/20/mentors-champions-program/) and [express your interest](https://airtable.com/shrlrLdQbb4wphosK) by __Monday November 7, 2022__.

If you have any questions, please, contact our Community Manager by [email](mailto:yabellini@ropensci.org) or [book a 15 minutes meeting with her](https://calendly.com/yabellini-ropensci/15min).

We will have a [Community Call on October 24th](/commcalls/oct2022-champions/)to explain and explore the benefits of Champions Programs and a [Coworking slot on November 1st](/events/coworking-2022-11/) to answer any questions you might have about mentoring in this program.
```

Reviewer 2: The blog post is merged but is not shown on the webpage.
